libraries/plugins/witness/include/scorum/witness/witness_plugin.hpp

```diff
@@ -79,10 +80,11 @@ class witness_plugin : public scorum::app::plugin
     virtual void plugin_shutdown() override;
 
 private:
-    void schedule_production_loop();
-    block_production_condition::block_production_condition_enum block_production_loop();
-    block_production_condition::block_production_condition_enum maybe_produce_block(
-        fc::mutable_variant_object& capture);
+    void schedule_production_loop(block_production_condition::block_production_condition_enum last_result
```

Comment:
schedule_production_loop does not depend on last_result, than why we blur interface?